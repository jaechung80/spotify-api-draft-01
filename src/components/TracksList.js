import React, { useEffect } from 'react';
import { connect } from 'react-redux';

const TracksList = (props) => {
  useEffect(() => {
    console.log('in TracksList props.tracks', props.tracks);
  });

  return (
    <div>
      {props.tracks?.items?.map((track) => {
        return (
          <ul key={track.id}>
            <li>Track Title: {track.name}</li>
            <li>Artist: {track.artists[0].name}</li>
            <li>Album: {track.album.name}</li>
            <li>
              <img
                src={
                  track.album.images.length > 0
                    ? track.album.images[0].url
                    : 'https://upload.wikimedia.org/wikipedia/commons/2/26/512pxIcon-sunset_photo_not_found.png'
                }
              />
            </li>
          </ul>
        );
      })}
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    tracks: state.tracks,
  };
};

export default connect(mapStateToProps)(TracksList);
