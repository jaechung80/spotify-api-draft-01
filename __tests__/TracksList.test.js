import React from 'react';
import TracksList from '../src/components/TracksList';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({ adapter: new Adapter() });

describe('TracksList component', () => {
  const wrapper = shallow(<TracksList />);

  it('renders TracksList component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
