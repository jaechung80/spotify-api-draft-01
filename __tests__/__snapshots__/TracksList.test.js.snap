// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TracksList component renders TracksList component correctly 1`] = `ShallowWrapper {}`;
